File: app_musik/app_musik_back/src/main/java/bj/highfiveuniversity/app_musik/models/Video.java
package bj.highfiveuniversity.app_musik.models;

import java.time.LocalDateTime;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "video")
public class Video {
    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String director;

    @Column(nullable = false)
    private String genre;

    @Column(nullable = false)
    private String duration;

    @Column(nullable = false)
    private String file;

    @Column(nullable = false)
    private String resolution;

    @Column(updatable = false, nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    // S'exécute quand on insère une donnée
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    // S'exécute quand on modifie une donnée
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: app_musik/app_musik_front/lib/videos/models/video.dart
class Video {
  final String title;
  final String director;
  final String genre;
  final int duration;
  final String file;
  final String resolution;
  final String createdAt;
  final String updatedAt;

  Video({
    required this.title,
    required this.director,
    required this.genre,
    required this.duration,
    required this.file,
    required this.resolution,
    required this.createdAt,
    required this.updatedAt,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'director': director,
      'genre': genre,
      'duration': duration,
      'file': file,
      'resolution': resolution,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }

  factory Video.fromJson(Map<String, dynamic> json) {
    return Video(
      title: json['title'],
      director: json['director'],
      genre: json['genre'],
      duration: json['duration'],
      file: json['file'],
      resolution: json['resolution'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
    );
  }
}
